opal/corelib/range.rb

```diff
@@ -126,18 +146,55 @@ def size
   end
 
   def step(n = 1)
-    raise NotImplementedError
+    return enum_for(:step, n) unless block_given?
+    raise ArgumentError.new("step can't be negative") if n < 0
+    raise ArgumentError.new("step can't be 0") unless n > 0
+    if @begin.is_a?(Numeric) and @end.is_a?(Numeric)
+      i = 0
+      loop do
+        current = @begin + i * n
+        if @exclude
+          break if current >= @end
+        else
+          break if current > @end
+        end
+        yield(current)
+        i += 1
+      end
+    else
+      raise TypeError, "step must be an integer" unless n.is_a?(Numeric) && n.to_int == n
```

Comment:
You can use `n = Opal.coerce_to(n, Integer, :to_int)` here (I guess you will get some passing specs in `spec/ruby` after this change)